NTupleMaker/interface/functionsCP.h

```diff
@@ -34,215 +35,184 @@ TLorentzVector gen_ipVec(const AC1B * analysisTree, int tauIndex, int piIndex, T
+    otree->VxConstitTau2=analysisTree->tau_constituents_vx[tauIndex2][piIndexfortau2];
+    otree->VyConstitTau2=analysisTree->tau_constituents_vy[tauIndex2][piIndexfortau2];   
+    otree->VzConstitTau2=analysisTree->tau_constituents_vz[tauIndex2][piIndexfortau2];      
+  }
+    
   
-  bool firstNegative = false;
-  if (analysisTree->tau_charge[tauIndex1]<0.0)
+  bool firstNegative = false;//Merijn 2019 1 10: instead I ask if the second is positive. WON'T WORK FOR E-MU CASE!
+//comment 2019 2 5: I need to re-evaluate why I implemented this. I think it is due to fact that initially lepton and muon index were innertwined
+  if (analysisTree->tau_charge[tauIndex2]>0.0)
     firstNegative = true;
 
   TLorentzVector tau1IP;
-//Merijn: for leptonic decay calculate in different way than for hadronic
+  //Merijn: for leptonic decay calculate in different way than for hadronic
   if(channel=="et"||channel=="mt") tau1IP = ipVec_Lepton(analysisTree,tauIndex1,channel);
-  else{tau1IP = ipVec(analysisTree,tauIndex1);}//tt
+  else{tau1IP = ipVec(analysisTree,tauIndex1);}//tt: treat as tau index..
   TLorentzVector tau1Pi0;
   tau1Pi0.SetXYZT(0.,0.,0.,0.);
-  
+
+  //Merijn: have to assume here we WON'T look into e-mu case! 
   TLorentzVector tau2IP;
   tau2IP = ipVec(analysisTree,tauIndex2);
   TLorentzVector tau2Pi0;
-  tau2Pi0.SetXYZT(0.,0.,0.,0.);
+  tau2Pi0.SetXYZT(0.,0.,0.,0.); 
 
   //Merijn: here calculate neutral pion vectors. Only look for pions in first tau if channel is tt.
-  if(channel=="tt"){ if (analysisTree->tau_decayMode[tauIndex1]>=1&&analysisTree->tau_decayMode[tauIndex1]<=3) tau1Pi0 = neutralPivec(analysisTree,tauIndex1);}
-  if (analysisTree->tau_decayMode[tauIndex2]>=1&&analysisTree->tau_decayMode[tauIndex2]<=3) tau2Pi0 = neutralPivec(analysisTree,tauIndex2);
+  if(channel=="tt"){ if(analysisTree->tau_decayMode[tauIndex1]>=1&&analysisTree->tau_decayMode[tauIndex1]<=3) tau1Pi0 = neutralPivec(analysisTree,tauIndex1);}
+
+  if (analysisTree->tau_decayMode[tauIndex2]>=1&&analysisTree->tau_decayMode[tauIndex2]<=3){ 
+	tau2Pi0 = neutralPivec(analysisTree,tauIndex2);
+  }
   
-//here can proceed with CP calculation
-  otree->acotautau_00=acoCPCOUT(tau1Prong,tau2Prong,tau1IP,tau2IP,firstNegative,false,false); // cout<<"otree->acotautau_00 "<<otree->acotautau_00<<endl;
 
-  if (analysisTree->tau_decayMode[tauIndex1]==1&&analysisTree->tau_decayMode[tauIndex2]==0)
-    otree->acotautau_10=acoCP(tau1Prong,tau2Prong,tau1Pi0,tau2IP,firstNegative,true,false);
+  otree->acotautau_00=acoCP(tau1Prong,tau2Prong,tau1IP,tau2IP,firstNegative,false,false,otree);
+   //I think it should work since everything assigned for 3 cases. Note: aco_00 will be filled with mu x 1-prong and mu x 1.1-prong
 
-  if (analysisTree->tau_decayMode[tauIndex1]==0&&analysisTree->tau_decayMode[tauIndex2]==1)
-    otree->acotautau_01=acoCP(tau1Prong,tau2Prong,tau1IP,tau2Pi0,firstNegative,false,true);
+  
+  if(channel=="et"||channel=="mt"){//we only fill aco_01 for et and mt
```

Comment:
Correct, however you also have to consider that the other variables wouldn't be filled for mt and et channels, given that the boolean requiring a decay with one prong+pi0s would have been false for the first tau decay.
I do still prefer to have the selection of the variables using the division in 1 prong / lepton, 1 prong + pi0s, 3 prongs but both ways are fine.

Reply:
Hi Andrea,
I think I implemented this at the time that the lepton and tau index were innertwined, which lead to some confusion.
If the effect is equal, perhaps leave for now. We can later revert if needed.

Reply:
Ok, now I understand, you used the check to find out that the indices of the two taus (or hadronic tau and lepton) were switched. This could also be useful in the future to remember that the two indices have to be put in the correct order, and at reco level the decay mode with 1 prong / lepton has the same code.
I had already added a comment in SynchNTuple producer stating that but this works as a second check.